### PR TITLE
chore: Expose ClickHouse logs in `docker-compose.dev.yml`

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

I've occasionally been running into ClickHouse and Kafka init errors in my local environment. This was hard to debug and I've always had to do Ctrl+C and re-run `docker compose up`, which took time.
 
## Changes

Let's restart ClickHouse and Kafka automatically on error, and expose ClickHouse logs.

### CODE REVIEW

File: docker-compose.dev.yml [hunk 1]
```diff
@@ -53,6 +56,7 @@ services:
             service: kafka
         ports:
             - '9092:9092'
+        restart: on-failure
```
Comment (hazzadous) [inline docker-compose.dev.yml:59]: Looks like we already have restart on failure set in the base docker compose from which we extend, does this not carry over?

**Review by hazzadous — APPROVED** [review]
Looks reasonable. I guess having the logs not just in stdout is to allow you to easily grep through the logs?

Unsure why the services don't already restart on failure, but fine to have it explicitly specified.

File: docker-compose.dev.yml (hunk 1 above)
Comment (Twixes) [reply docker-compose.dev.yml]: Weird, that didn't seem like the case for me before, but maybe I overlooked the restarts 🤔 Reverted this change

**Comment by Twixes** [comment]
About the logs, the primary purpose is debugging "ClickHouse init process failed", as for that error the relevant information is only in those files and not in stdout.

**Comment by Twixes** [comment]
Weirdly this benign change causes `docker compose up` to stop working in CI now. 🤔 Don't wanna spend time digging into this, so I'll just close this. Can always add these lines manually when the need arises.

**Pull request closed without merging**